aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -296,7 +287,7 @@ impl Aggregator {
     }
 
     #[allow(dead_code)]
-    fn get_value(&self) -> PartialVMResult<u128> {
+    fn get_value(&self) -> BoundedMathResult<u128> {
```

Comment:
Just trying to understand good design patterns. What's the benefit of distinguishing BoundedMathResult from PartialVMResult here?

Reply:
not sure which result we should be returning here. 


but at the high level - PartiaulVMResult is uncostrained - anything error.

BoundedMathResult basically restricts errors to be only overflow and underflow.
If we have a numeric operation (like addition), that can only fail in those two ways, returning BoundedMathResult allows caller to be able to handle errors (i.e. it knows those are the only two outcomes, and can handle each the way it needs) - i.e. enabled things like expect_ok(ok_overflow())